src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -153,13 +153,20 @@ private static NativeMethods LoadNativeMethodsXamarin()
             return new NativeMethods(new NativeMethods.DllImportsFromStaticLib());
         }
 
-        private static string GetAssemblyPath()
+        private static string GetAssemblyDirectory()
         {
             var assembly = typeof(NativeExtension).GetTypeInfo().Assembly;
 #if NETSTANDARD
             // Assembly.EscapedCodeBase does not exist under CoreCLR, but assemblies imported from a nuget package
             // don't seem to be shadowed by DNX-based projects at all.
-            return assembly.Location;
+            var assemblyLocation = assembly.Location;
+            if (assemblyLocation != null)
```

Comment:
Actually, assembly.Location might be returning an empty string according to the docs:
https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file#api-incompatibility

I'll check which one is correct and update the code if needed.